rpc/http.go

```diff
@@ -39,7 +39,10 @@ const (
 var acceptedContentTypes = []string{contentType, "application/json-rpc", "application/jsonrequest"}
 
 type httpConn struct {
+	sync.Mutex
+
 	client    *http.Client
+	headers   map[string]string
```

Comment:
Might be better to use the `http.Header` type for this. It keeps the keys lowercase.